compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1018,8 +1018,14 @@ object Types {
      */
     def safe_& (that: Type)(using Context): Type = (this, that) match {
       case (TypeBounds(lo1, hi1), TypeBounds(lo2, hi2)) =>
-        TypeBounds(OrType(lo1.stripLazyRef, lo2.stripLazyRef), AndType(hi1.stripLazyRef, hi2.stripLazyRef))
-      case _ => this & that
+        def safeOrType(tp1: Type, tp2: Type): Type =
+          ctx.typeComparer.liftIfHK(tp1, tp2, OrType(_, _), safeOrType, _ & _)
+        def safeAndType(tp1: Type, tp2: Type): Type =
```

Comment:
I think it's better to roll liftIfHK into methods OrType.safe and AndType.safe instead of adding the methods here.